algorithmForNN/pointCloudUtils.py

```diff
@@ -0,0 +1,198 @@
+    points = get_points(plane_model, np_array, min_value)
+
+    if len(points) < 3:
+        return 0
+
+    result_area = 0
+    
+    for combination in get_combination(len(points), 3):
+        first_point = points[combination[0]]
+        second_point = points[combination[1]]
+        third_point = points[combination[2]]
+        
+        current_area = get_S_triangle(first_point, second_point, third_point)
+        
+        if current_area > result_area: result_area = current_area
+    
+    return result_area
+
+def segment_all_planes_from_point_cloud(point_cloud : o3d.geometry.PointCloud, min_count_of_points : int, min_area_of_plane : int, distance : float = 0.1) -> list:
+    '''
+    Function allows you to get all planes from a given point cloud 
+    param: distance - Max distance a point can be from the plane model, and still be considered an inlier.
+    '''
+    all_planes = []
+    inlier_cloud, outlier_cloud, plane_model = segment_plane_from_point_cloud(point_cloud, distance)
+    while outlier_cloud.has_points():
+        if len(convert_point_cloud_to_numpy_array(inlier_cloud)) > min_count_of_points and get_area_of_plane(plane_model, convert_point_cloud_to_numpy_array(inlier_cloud)) > min_area_of_plane:
+            all_planes.append(inlier_cloud)
+        inlier_cloud, outlier_cloud, plane_model = segment_plane_from_point_cloud(outlier_cloud, distance)
+    if inlier_cloud.has_points() and len(convert_point_cloud_to_numpy_array(inlier_cloud)) > min_count_of_points and get_area_of_plane(plane_model, convert_point_cloud_to_numpy_array(inlier_cloud)) > min_area_of_plane: all_planes.append(inlier_cloud)
+    
+    return all_planes
+
+def create_dictionary_of_point_cloud(point_cloud : o3d.geometry.PointCloud) -> dict:
+    '''
+    Function returns a dictionary of points, with which you can get indices faster 
+    '''
+    result_dictionary = {}
+    numpy_main_point_cloud = convert_point_cloud_to_numpy_array(point_cloud)
+    
+    for idx, point in enumerate(numpy_main_point_cloud):
+        triple = (point[0], point[1], point[2])
+        result_dictionary[triple] = idx
+        
+    return result_dictionary
+
+def get_indexes_of_points(current_dict : dict, temp_point_cloud : np.ndarray) -> list:
+    '''
+    Function finds the indices of the required points 
+    '''
+    index_list = []
+    for point in temp_point_cloud:
+        triple = (point[0], point[1], point[2])
+        index_list.append(current_dict[triple])
+    return index_list
+
+def create_label_file(current_snapshot : str, path_to_label_file : str, label_list : list, path_to_new_label_file : str, min_count_of_points : int, min_area_of_plane : int):
+    '''
+    Function creates a file with binary labels 
+    '''
+    main_point_cloud = get_point_cloud_from_bin_file(current_snapshot)
+    numpy_main_point_cloud = convert_point_cloud_to_numpy_array(main_point_cloud)
+    write_zeros_in_file(len(numpy_main_point_cloud), path_to_new_label_file)
```

Comment:
Why it is important to write 0s to file first? Is it not enough memory to perform all in memory and then write to file only result?

Reply:
Yes, I know that because of this the speed of the algorithm drops. At the moment, my goal is to get a working algorithm. YV Litvinov says that it is bad practice to immediately try to speed up programs and algorithms. The same applies to unnecessary conversions from numpy array to point cloud. I think in some places this can be avoided, but I think it will be done after the algorithm works correctly.